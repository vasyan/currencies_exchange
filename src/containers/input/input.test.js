import React from 'react'
import { shallow, mount } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import configureMockStore from 'redux-mock-store'
import Input from './index'

describe('<Input>', () => {
  const store = configureMockStore()({
    widgets: {
      exchange: {
        currencyFrom: 'FOO',
        currencyTo: 'BAR',
        amount: '42'
      }
    },
    collections: {
      rates: {
        USD: {
          FOO: 0.4242,
          BAR: 0.2424
        }
      }
    }
  })
  it('should render', () => {
    expect(
      shallowToJson(shallow(<Input store={store} />).dive())
    ).toMatchSnapshot()
  })

  it('should call action on currency input change', () => {
    const props = {
      onInputChange: jest.fn()
    }

    const wrapper = mount(<Input store={store} {...props} />)
    const event = { target: { value: '42.00' } }

    wrapper.find('input').simulate('change', event)

    expect(store.getActions()).toMatchSnapshot()
  })
})
