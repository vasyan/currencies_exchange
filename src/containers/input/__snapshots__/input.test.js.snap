// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input> should call action on currency input change 1`] = `
Array [
  Object {
    "payload": "42",
    "type": "exchange/AMOUNT_INPUT",
  },
]
`;

exports[`<Input> should render 1`] = `
<div
  className="wrapper"
>
  <CurrencySelect
    currencyName="FOO"
    onNext={[Function]}
    onPrev={[Function]}
  >
    <div
      className="input_wrapper"
    >
      <input
        autoFocus={true}
        className="input"
        onChange={[Function]}
        type="text"
        value="42"
      />
    </div>
  </CurrencySelect>
</div>
`;
